### PR TITLE
ST-4278 Added migration yaml to allow upload to PaaS

### PR DESCRIPTION
**Pull request by pierre-noumena**

### CODE REVIEW

**Comment by ari-noumena** [comment]
Isn't this ST-4278?

File: src/main/yaml/migration.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+systemUnderAudit: nplstarter
+
+changesets:
+  - name: 1.0
+    changes:
+      - reset: true
+      - npl-load:
+        dir-list: npl
```
Comment (ari-noumena) [inline src/main/yaml/migration.yml:8]: Nit: when Github shows this little symbol in the diff, it means that there is no newline at the end of the file. Generally our convention is that all files should end with a newline (but it's not an important one).

image

File: src/main/npl-1.0/objects/iou/iou.npl [hunk 2]
```diff
@@ -38,6 +39,11 @@ protocol[issuer, payee] Iou(var forAmount: Number) {
         become forgiven;
     };
 
+    @api
+    permission[payee] confirmRepaid() | repaid {
+        become confirmed_repaid;
+    };
```
Comment (ari-noumena) [inline src/main/npl-1.0/objects/iou/iou.npl:45]: These are behavioural changes that weren't specified in the ticket... Are they necessary?

Once (if) we've made changes to the obligation semantics, we'll revisit these examples and presumably use obligations correctly. But for now I'd leave out any changes unless there is a good reason to introduce them. This diff would be a perfect example of some business logic to add in a demo, which can then be deployed in order to show the new API being generated.

Comment (pierre-noumena) [reply src/main/npl-1.0/objects/iou/iou.npl]: The content changes to the IoU file were uninted, thanks for noticing, should have checked the diff. Fixed

File: src/main/yaml/migration.yml (hunk 1 above)
Comment (pierre-noumena) [reply src/main/yaml/migration.yml]: Fixed

**Review by ari-noumena — APPROVED** [review]

**Pull request merged**